### PR TITLE
Process blind withdrawals

### PR DESCRIPTION
**Pull request by terencechain**
With a blind payload (i.e header), the withdrawal list is represented as a root. Given limited information, a node will not be able to process the state root using a withdrawal root. The node should use expected withdrawals in the state to process withdrawal when processing a blind payload.

### CODE REVIEW

File: beacon-chain/core/blocks/withdrawals.go [hunk 1]
```diff
@@ -110,47 +112,71 @@ func ValidateBLSToExecutionChange(st state.ReadOnlyBeaconState, signed *ethpb.Si
...
+	if err != nil {
+		return nil, errors.Wrap(err, "could not get expected withdrawals root")
+	}
+	if expectedRoot != wdRoot {
+		return nil, fmt.Errorf("expected withdrawals root %#x, got %#x", expectedRoot, wdRoot)
 	}
-	for i, withdrawal := range withdrawals {
-		if withdrawal.Index != expected[i].Index {
+
+	for i, withdrawal := range expectedWithdrawals {
```
Comment (nisdas) [inline beacon-chain/core/blocks/withdrawals.go:147]: This doesn't make any sense, you are comparing the same objects in each loop . Basically if `a.Index != a.Index` , which will never be triggered.

**Comment by rkapka** [comment]
Can you add a unit test to blinded/non-blinded cases where the root doesn't match the expected root?

**Comment by terencechain** [comment]
> 

Done

**Review by rauljordan — COMMENTED** [review]
Hi @terencechain , @saolyn and I just went through this together and the main feedback we have is that this PR should have a clear description of the changes and also why we need them. Also, a comment on ProcessWithdrawals that mentions this will work with both blinded and full execution data would be ideal. We also wanted to double check that both branches of the conditional added are covered in tests

**Review by nisdas — APPROVED** [review]

**Review by nisdas — APPROVED** [review]

**Review by kasey — APPROVED** [review]

**Pull request merged**